federatedscope/cl/loss/NT_xentloss.py

```diff
@@ -0,0 +1,41 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from federatedscope.register import register_criterion
+
+class NT_xentloss(nn.Module):
+    def __init__(self, temperature=0.5):
+        super(NT_xentloss, self).__init__()
+        self.temperature = temperature
+        
+    def forward(self, z1, z2):
+        N, Z = z1.shape 
+        device = z1.device 
+        representations = torch.cat([z1, z2], dim=0)
+        similarity_matrix = F.cosine_similarity(representations.unsqueeze(1), representations.unsqueeze(0), dim=-1)
+
+        l_pos = torch.diag(similarity_matrix, N)
```

Comment:
I think, up to now, `similarity_matrix` is a 2N-by-2N matrix. Am I wrong? Why do we need to take the the above and below main diagonal?